src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -165,6 +168,17 @@
             Ref("NumericLiteralSegment"),
         ),
     ),
+    QuotedLiteralSegment=AnyNumberOf(
+        # MySQL allows whitespace-concatenated string literals (#1488).
+        # Since these string literals can have comments between them,
+        # we use grammar to handle this.
+        NamedParser(
+            "single_quote",
+            CodeSegment,
+            name="quoted_literal",
+            type="literal",
+        ),
+    ),
```

Comment:
MySQL is very simple: Since it's string literals can be whitespace and/or newline spaced it's just `AnyNumberOf`.